activerecord/test/cases/locking_test.rb

```diff
@@ -738,6 +738,29 @@ def test_with_lock_rolls_back_transaction
       assert_equal old, person.reload.first_name
     end
 
+    def test_with_lock_forwards_transaction_options
+      person = Person.find 1
+      assert_called_with(person, :transaction, [requires_new: true, isolation: "serializable", joinable: false]) do
```

Comment:
This is testing the implementation details. We avoid to do that in Rails. Can you change the test to really test if the transaction was open with the right options?

Reply:
Sounds good. I updated this test to introspect the resulting transaction.